### PR TITLE
Clarify the behavior of the `files` array

### PR DESCRIPTION
**Pull request by jbcpollak**
The `files` documentation is a bit confusing (and actually incorrect with respect to current `files` vs `.npmignore` behavior, so I hope this clarifies it.

### CODE REVIEW

**Review by coconutperm — APPROVED** [review]

File: doc/files/package.json.md
Comment (legodude17) [inline doc/files/package.json.md]: Move this paragraph down.

File: doc/files/package.json.md
Comment (legodude17) [inline doc/files/package.json.md]: Put the description of files down here.

**Review by legodude17 — CHANGES_REQUESTED** [review]
Nice PR!

(Note that I am not a developer of `npm`, just a community member)

File: doc/files/package.json.md
Comment (jbcpollak) [inline doc/files/package.json.md]: @legodude17 I am ok with the change you suggest, but was trying to minimize the documentation change under the theory that would be more acceptable to the core team. I'm happy to update if that is preferred.

Comment (legodude17) [inline doc/files/package.json.md]: I would say it makes more sense to merge the `Files included with the "package.json#files" filed....` and the `The "files" field....` paragraphs.

**Comment by iarna** [comment]
This is definitely an improvement (for future reference, you _can_ override `files` with a `.gitignore` or `.npmignore` in a _subdirectory_, but not at the top level—while this'd be good to document, I don't want to hold up this PR any longer. There's no need for the perfect, we can always do another pass.) 

I've merged  this as a0fdd1571. Thank you for your patience.

**Pull request closed without merging**
